ably.ts.d

```diff
@@ -0,0 +1,457 @@
+
+    get: (waitForSync: boolean, clientId?: String, connectionId?: String) => PresenceMessage;
+
+    history: (start: number, end: number, direction: string, limit: number, untilAttach: boolean) => PaginatedResult;
+
+    subscribe: (presense: PresenceMessage | PresenceAction, presenceMessage?: PresenceMessage) => void;
+
+    unsubscribe: (presense?: PresenceMessage | PresenceAction, presenceMessage?: PresenceMessage) => void;
+
+    enter: (Data: any) => void;
+
+    update: (Data: any) => void;
+
+    leave: (Data: any) => void;
+
+    enterClient: (clientId: String, Data: any) => void;
+
+    updateClient: (clientId: String, Data: any) => void;
+
+    leaveClient: (clientId: String, Data: any) => void;
+  }
+
+  type MessageFn = (msg: Message) => void;
+
+  export declare class RestChannel {
+    name: String;
+    presence: RestPresence;
+    history: (start: number, // TIME
+              end: number, // time
+              direction: String,
+              limit: number) => PaginatedResult;
+
+    publish: (name?: Array<String> | String, data?: any, clientId?: String, extras?: any) => any;
+  }
+
+  export declare class RealtimeChannel extends EventEmitter {
+    name: String;
+    errorReason: ErrorInfo;
+    state: ChannelState;
+    presence: RealtimePresence;
+
+    attach: () => any;
+
+    detach:() => any;
+
+    history: (start: number,
+            end: number,
+            direction: String,
+            limit: number,
+            untilAttach: boolean) => PaginatedResult;
+
+    subscribe: (Msg: String | Message | MessageFn, ExtraMsg?: Message) => any;
+
+    unsubscribe: (Msg?: String | Message | MessageFn, ExtraMsg?: Message) => any;
+
+    publish: (name?: Array<String> | String, data?: any, clientId?: String, extras?: any) => any;
+
+  }
+
+  export declare class Channels {
+    exists: (name: String) => boolean;
+
+    get: (name: String, channelOptions?: ChannelOptions) => RealtimeChannel;
+    iterate: () => RealtimeChannel;
+
+    release: (name: String) => void;
+  }
+
+  export declare class Message {
+    constructor(name?: String, data?: any, clientId?: String);
+
+    fromEncoded: (JsonObject: String, channelOptions: ChannelOptions) => Message;
+
+    fromEncodedArray: (JsonArray: String, channelOptions: ChannelOptions) => Message;
+
+    clientId: String;
+    connectionId: String;
+    data: any;
+    encoding: String;
+    extras: String;
+    id: String;
+    name: String;
+    timestamp: number; // Time
+  }
+
+  export declare class PresenceMessage {
+    fromEncoded: (JsonObject: any, channelOptions?: ChannelOptions) => PresenceMessage;
+
+    fromEncodedArray: (JsonArray: Array<any>, channelOptions?: ChannelOptions) => Array<PresenceMessage>;
+
+    action: PresenceAction;
+    clientId: String;
+    connectionId: String;
+    data: any;
+    encoding: String;
+    id: String;
+    timestamp: number; // Time
+    memberKey: () => String;
+  }
+
+  export declare class ProtocolMessage {
```

Comment:
I was going off the IDL to make the typings.  Anything in the IDL was translated.   If nothing uses the ProtocolMessage then it is pointless to have in the Typings.